File: Study Plan/Algorithm I/Squares_Sorted_Array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int start = 0, len = nums.size(), end = len - 1;
        vector<int> result(len);
        while(start <= end)
        {
            if(abs(nums[start]) > abs(nums[end]))
            {
                result[len - 1] = nums[start] * nums[start];
                start ++;
            }
            else
            {
                result[len - 1] = nums[end] * nums[end];
                end --;
            }
            len --;
        }
        return result; 
    }
};

File: Study Plan/Algorithm I/Squares_Sorted_Array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        N = len(nums)
        start, end = 0, N - 1
        result = [0] * N
        while start <= end:
            if abs(nums[start]) > abs(nums[end]):
                result[N - 1] = nums[start] * nums[start]
                start += 1
            else:
                result[N - 1] = nums[end] * nums[end]
                end -= 1
            N -= 1
        return result
        